### PR TITLE
feat: add tools for unit test

### PR DESCRIPTION
**Pull request by caitong93**
**What this PR does / why we need it**:

Add helper functions for writing unit test.

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @kdada @ddysher 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: utils/unittest/unittest.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+func (r *responseWriter) Bytes() []byte {
+	return r.buf.Bytes()
+}
+
+// NewTestService creates a service.Service for testing
+func NewTestService(desc ...definition.Descriptor) (service.Service, error) {
+	builder := service.NewBuilder()
+	builder.SetModifier(service.FirstContextParameter())
+	builder.AddFilter(service.RedirectTrailingSlash(), service.FillLeadingSlash(), service.ParseRequestForm())
```
Comment (kdada) [inline utils/unittest/unittest.go]: Modifiers and filters may be changed by users. So these configs needs to be exported.

File: utils/unittest/unittest.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+// NewJSONRequest creates a http.Request with json Content-Type. The data parameter can be io.Reader,
+// []bytes or a struct.
+func NewJSONRequest(ctx context.Context, method, url string, data interface{}) (*http.Request, error) {
+	var r io.ReadCloser
+	if data != nil {
+		switch t := data.(type) {
+		case io.ReadCloser:
+			r = t
+		case io.Reader:
+			r = ioutil.NopCloser(t)
```
Comment (kdada) [inline utils/unittest/unittest.go]: Why do you wrap reader?

File: utils/unittest/unittest.go (hunk 1 above)
Comment (caitong93) [reply utils/unittest/unittest.go]: added a new function `NewTestServiceWithConfig `

**Comment by caitong93** [comment]
@kdada updated

**Comment by kdada** [comment]
/lgtm

This package needs more utils for testing nirvana.

**Pull request merged**